### PR TITLE
Gardener Buffs Again

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Added several new fun features to Gardener drones, such as:

- Six Fruit Slots
- Hardy Weeds
  - These weeds can be planted on semi-weedable turfs, such as jungle grass and big red mars turfs.
  - Planting a node has a two minute cooldown, so you better defend it from marines.
  - They have 10 HP, which is a level between normal weeds (5) and hive weeds (15).
  - Cannot support special structures, such as hive cores, clusters, recovery nodes, etc.
  - Let any fruits planted on them grow in 80% of the usual time.
- Queen Fruit Share
  - When you feed a queen, if she's observing a xeno and the xeno is on the same z-level as her, the observed xeno and the queen both benefit from the fruit.
- Cruor Fruits
  - These fruits boost the plasma regen of xenos that consume them, granting them a total of 240 plasma over 15 seconds.
  - It takes 25 seconds to grow.
- Resin Surge can now be used while resting, and on any visible turf. If you click on a mob, it'll use the old line-cast system to see if you can use it.

Additionally, shuffled some of the macros for Gardener around:
- 1: Plant Weeds
- 2: Plant Hardy Weeds
- 3: Select Resin Surge
- 4: Plant Fruit

## Why It's Good For The Game
Gardener drones leave a lot to be desired still, with the previous buffs only managing to make them 'not terrible'. The biggest change here is the hardy weeds, which allows xenos to stage battles from new areas of maps previously impossible due to the weed limitations. The limit of one node every 2 minutes, with marines easily being able to dispose of the node with a grenade or flamer, shouldn't make it too much of an issue.

## Changelog

:cl:
balance: Gardener drones now have six fruit slots.
add: Gardener drones can now plant hardy weeds in addition to normal weeds. Hardy weeds have a two minute cooldown, but can grow on grass and red sand turfs, and let fruits grow in 80% of the usual time. They cannot support special structures, and have a bit more health than standard weeds.
add: Feeding fruit to a queen while she's overwatching a xeno on the same z-level now grants the fruit benefit to that xeno in addition to the queen.
add: Added cruor fruits, which grants the xeno who eats it 240 plasma over 15 seconds.
tweak: Resin surge can now be used while resting, and works on any visible turf.
/:cl:

### CODE REVIEW

**Comment by Bram-KEP** [comment]
Based

**Comment by Listaless** [comment]
Queen fruit share will likely lead to the return of pocket healing/buffing.
also I'd like people to remember that healer drone exists and Gardener outperforms it and will do more so with six fruit slots.

**Comment by c4xmaniac** [comment]
The queen fruit sharing is super queen healing which doesn’t need a buff.

**Comment by Dethstorm** [comment]
Coolest of beans, but Hardy Weeds feels like a weird name. That aside, amazing stuff

**Comment by RadiationFan** [comment]
An idea for the coder, if you will hear me. 
Perhaps consider giving the hardy weeds something like the old techweb resistance, nothing crazy like 50% damage reduction, but perhaps a 5-15% reduction, enough that it feels worthwhile to fight for and hold, perhaps even make the reduction scale up over time, so that defending and attack feels like a significant mission for both sides. Something like a five minute windup, if the node dies the timer ends and the resistances are removed.

**Comment by Nanu308** [comment]
Cool concepts that can be tweaked further down the line

File: code/modules/cm_aliens/structures/fruit.dm [hunk 1]
```diff
@@ -283,11 +289,34 @@
...
+	flags = CAN_CONSUME_AT_FULL_HEALTH
+	fruit_type = /obj/item/reagent_container/food/snacks/resin_fruit/plasma
+	glow_color = "#287A90"
+	var/plasma_amount = 240
+	var/plasma_time = 15
+	var/time_between_plasmas = 3
+
+/obj/effect/alien/resin/fruit/plasma/consume_effect(mob/living/carbon/Xenomorph/recipient, var/do_consume = TRUE)
+	if(mature && recipient && recipient.plasma_max > 0 && !QDELETED(recipient))
+		to_chat(recipient, SPAN_XENONOTICE("The [name] boosts your plasma regeneration!"))
```
Comment (stanalbatross) [inline code/modules/cm_aliens/structures/fruit.dm:315]:
```suggestion
		to_chat(recipient, SPAN_XENONOTICE("\The [name] boosts your plasma regeneration!"))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/cm_aliens/structures/fruit.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/cm_aliens/structures/fruit.dm:315]: [name] is a string, not a mob, so \the wouldn't work with it

Comment (stanalbatross) [reply code/modules/cm_aliens/structures/fruit.dm:315]: wah

**Pull request merged**